MOBT-2-Swift-Foundation.playground/Contents.swift

```diff
@@ -0,0 +1,332 @@
+    case Romania = "Romania"
+    case Netherlands = "Netherlands"
+    
+    var configurationPath: String {
+        switch self {
+        case .Italy:
+            return "ConfigurationItaly"
+        case .Greece:
+            return "ConfigurationGreece"
+        case .Romania:
+            return "ConfigurationRomania"
+        case .Netherlands:
+            return "ConfigurationNetherlands"
+        }
+    }
+    
+    var localizedTableName: String {
+        switch self {
+        case .Italy:
+            return "LocalizedItaly"
+        case .Greece:
+            return "LocalizedGreece"
+        case .Romania:
+            return "LocalizedRomania"
+        case .Netherlands:
+            return "ConfigurationNetherlands"
+        }
+    }
+}
+
+/// This is the example of getting backend part of code. And transfer this data to swift objects.
+// MARK: - Typealias
+public typealias EmptyClosure = () -> Void
+public typealias DataKeys = (([String]) -> Void)
+public typealias DataValues = (([Double]) -> Void)
+
+public struct Keys {
+    static let mostRecentExchangeRatesValues = "Keys.mostRecentExchangeRatesValue"
+    static let mostRecentExchangeRatesKeys = "Keys.mostRecentExchangeRatesKeys"
+}
+
+// MARK: - Protocols
+protocol CurrencyExchangeMethods {
+    func fetchJSON(retreiveDataKeys: @escaping DataKeys, retrieveDataValues: @escaping DataValues)
+}
+
+// MARK: - Classes
+final class CurrencyExchangeConverterService: CurrencyExchangeMethods {
+    // MARK: - Enums
+    private enum Const {
+        static let urlAPI: String = "https://open.er-api.com/v6/latest/USD"
+    }
+    
+    // MARK: - Properties
+    var currencyCode: [String] = []
+    var values: [Double] = []
+    
+    // MARK: - Methods
+    func fetchJSON(retreiveDataKeys: @escaping DataKeys, retrieveDataValues: @escaping DataValues) {
+        guard let url = URL(string: Const.urlAPI) else { return }
+        URLSession.shared.dataTask(with: url) { (data, response, error) in
+            if error != nil { return }
+            guard let data = data else { return }
+            do {
+                let results = try JSONDecoder().decode(Currency.self, from: data)
+                self.currencyCode.append(contentsOf: results.rates.keys)
+                self.values.append(contentsOf: results.rates.values)
+                retreiveDataKeys(self.currencyCode)
+                retrieveDataValues(self.values)
+            } catch {
+                print(error)
+            }
+        }.resume()
+    }
+}
+
+final class CurrencyExchangeLocalData {
+    var currencyExchangeService: CurrencyExchangeConverterService?
+    
+    init(currencyExchangeService: CurrencyExchangeConverterService?) {
+        self.currencyExchangeService = currencyExchangeService
+    }
+    
+    func setDataToUserDefaults() {
+        currencyExchangeService?.fetchJSON(retreiveDataKeys: { keys in
+            UserDefaults.standard.set(keys, forKey: Keys.mostRecentExchangeRatesKeys)
```

Comment:
We should avoid using shared state, defaults should be injected via the initialiser